### ISSUE TITLE
Automatic value clamping is confusing and irrelevant for most use cases.

### ISSUE THREAD

**Issue by end2endzone** [end2endzone/libVariant#1]
The current implementation of the class prevents a user from getting an overflown value by clamping/saturating a value to its minimum or maximum value.

For example:
```cpp
Variant var;
var.setSInt16(32000);
var = var * 10; //forces type promotion to sint32 with a value of 320000
Variant::sint16 clamped_value = var.getSInt16(); //clamped_value = 32767
```

This implementation is justified to prevents array out of bounds issues (ie accessing `array[320000]`) by limiting access to the array. However, the default implementation (without using `Variant` class) would be to overflow the value (`clamped_value = -7680`). From this value, one can achieve the same result by limiting the value between minimum (0) and maximum (32767) values of an `sint16`.

The bounds validation should not be implemented in the Variant class but in the code which accesses the array.

**Comment by end2endzone** [comment]
The code is mostly handled by the `logicalConvert<>`, `getUnsignedRanges<>`, `getSignedRanges<>` and `staticCastConversion<>` templates.

**Comment by end2endzone** [comment]
The clamping value functionality is also irrelevant when the Variant is used to compute an array index. Most array size does not match the whole range of values for a given native type. For instance, it is not expected to have a `size_t` variable for addressing an array index when the valid index certainly does not range from 0 to 2,147,483,647.

**Comment by end2endzone** [comment]
Merged back to master in 384e7347b7b90cadfe154a88cf4cbb4571a6df4a
